frontend/pkg/frontend/frontend.go

```diff
@@ -147,6 +147,40 @@ func (f *Frontend) ArmResourceListBySubscription(writer http.ResponseWriter, req
 
 	f.logger.Info(fmt.Sprintf("%s: ArmResourceListBySubscription", versionedInterface))
 
+	subscriptionId := request.PathValue(PathSegmentSubscriptionID)
+
+	query := fmt.Sprintf("azure.subscription_id='%s'", subscriptionId)
+	clustersListResponse, err := f.conn.ClustersMgmt().V1().Clusters().List().Search(query).Send()
+	if err != nil {
+		f.logger.Error(err.Error())
+		arm.WriteInternalServerError(writer)
+		return
+	}
+
+	systemData := &arm.SystemData{} //Empty data until https://issues.redhat.com/browse/ARO-8002 is completed
+	var hcpCluster *api.HCPOpenShiftCluster
+	clusters := clustersListResponse.Items().Slice()
+	for _, cluster := range clusters {
+
+		hcpCluster, err = f.ConvertCStoHCPOpenShiftCluster(systemData, cluster)
+		if err != nil {
+			f.logger.Error(err.Error())
+			arm.WriteInternalServerError(writer)
+			return
+		}
+
+		versionedResource := versionedInterface.NewHCPOpenShiftCluster(hcpCluster)
+		resp, err := json.Marshal(versionedResource)
+		if err != nil {
+			f.logger.Error(err.Error())
+			arm.WriteInternalServerError(writer)
+			return
+		}
+		_, err = writer.Write(resp)
+		if err != nil {
+			f.logger.Error(err.Error())
+		}
```

Comment:
**Listing by subscription:**
`curl -X GET "localhost:8443/subscriptions/00000000-0000-0000-0000-000000000000/providers/Microsoft.RedHatOpenshift/hcpOpenShiftClusters?api-version=2024-06-10-preview"                 
{"id":"/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/klin-test-rg/providers/Microsoft.RedHatOpenShift/hcpOpenShiftClusters/klin-test-name","location":"eastus","name":"klin-test-name","properties":{"provisioningState":"","spec":{"api":{"url":"","visibility":"external"},"console":{"url":""},"disableUserWorkloadMonitoring":false,"dns":{"baseDomain":"","baseDomainPrefix":"klin-aro-hcp"},"etcdEncryption":false,"externalAuth":{"enabled":false,"externalAuths":[]},"fips":false,"ingress":[{"ip":"","url":"","visibility":""}],"issuerUrl":"","network":{"hostPrefix":23,"machineCidr":"10.0.0.0/16","networkType":"OVNKubernetes","podCidr":"10.128.0.0/14","serviceCidr":"172.30.0.0/16"},"platform":{"etcdEncryptionSetId":"","managedResourceGroup":"klin-test-mrg","networkSecurityGroupId":"/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/klin-test-rg/providers/Microsoft.Network/networkSecurityGroups/myNSG","outboundType":"loadBalancer","subnetId":"/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/klin-test-rg/providers/Microsoft.Network/virtualNetworks/vnet/subnets/mySubnet"},"proxy":{"httpProxy":"","httpsProxy":"","noProxy":"","trustedCa":""},"version":{"availableUpgrades":[],"channelGroup":"stable","id":"openshift-v4.15.11"}}},"systemData":{"createdBy":"","createdByType":"","lastModifiedBy":"","lastModifiedByType":""},"tags":{},"type":"Microsoft.RedHatOpenShift/hcpOpenShiftClusters"}{"id":"/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/klin-test-rg-2/providers/Microsoft.RedHatOpenShift/hcpOpenShiftClusters/klin-test-name-2","location":"eastus","name":"klin-test-name-2","properties":{"provisioningState":"","spec":{"api":{"url":"","visibility":"external"},"console":{"url":""},"disableUserWorkloadMonitoring":false,"dns":{"baseDomain":"","baseDomainPrefix":"klin-aro-hcp-2"},"etcdEncryption":false,"externalAuth":{"enabled":false,"externalAuths":[]},"fips":false,"ingress":[{"ip":"","url":"","visibility":""}],"issuerUrl":"","network":{"hostPrefix":23,"machineCidr":"10.0.0.0/16","networkType":"OVNKubernetes","podCidr":"10.128.0.0/14","serviceCidr":"172.30.0.0/16"},"platform":{"etcdEncryptionSetId":"","managedResourceGroup":"klin-test-mrg","networkSecurityGroupId":"/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/klin-test-rg-2/providers/Microsoft.Network/networkSecurityGroups/myNSG","outboundType":"loadBalancer","subnetId":"/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/klin-test-rg-2/providers/Microsoft.Network/virtualNetworks/vnet/subnets/mySubnet"},"proxy":{"httpProxy":"","httpsProxy":"","noProxy":"","trustedCa":""},"version":{"availableUpgrades":[],"channelGroup":"stable","id":"openshift-v4.15.11"}}},"systemData":{"createdBy":"","createdByType":"","lastModifiedBy":"","lastModifiedByType":""},"tags":{},"type":"Microsoft.RedHatOpenShift/hcpOpenShiftClusters"}%  `

Reply:
We will need to assemble our list of clusters into an array and then pass it into this type: https://github.com/Azure/ARO-HCP/blob/262bbec6c3a6327dc03ea901d578082bec311e2e/internal/api/v20240610preview/generated/models.go#L313-L320
and write it out once instead of writing out all the clusters appended to each other

Reply:
>and write it out once instead of writing out all the clusters appended to each other

We need to consider paging too, though, in case the result list is large.  [ARM defines is a bit of a protocol for this](https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-pageable), and it's complex enough that I think it's worth deferring to a separate card.  ([ARO-RP is a good implementation reference](https://github.com/Azure/ARO-RP/blob/master/pkg/frontend/openshiftcluster_list.go))

As a 1st pass, though, I agree just stuff it all in the Value slice above.

Reply:
Created https://issues.redhat.com/browse/ARO-8827 as a followup for pagination